lib/child_process.js

```diff
@@ -347,7 +347,11 @@ function normalizeSpawnArguments(file /*, args, options*/) {
     }
   }
 
-  args.unshift(file);
+  if (typeof options.argv0 === 'string') {
```

Comment:
The `argv0` here isn't necessarily from `process.argv0`, it's a parameter passed by the user. I think the check is still necessary.